### ISSUE TITLE
InvalidOptionKey error when using different backends

### ISSUE THREAD

**Issue by kevynlebouille** [shioyama/mobility#492]
Hi Chris,

I ran into an Mobility::Plugins::Backend::InvalidOptionKey error when I use container backend

## Context
In my mobility.rb initializer

`backend :jsonb, column_suffix: '_translations'`

In my content.rb model

```
class Content < ApplicationRecord
  # t.jsonb :custom_fields
  Mobility.available_locales.each { |locale| store_accessor :custom_fields, "vimeo_url_#{locale}" }

  # Translates
  extend Mobility
  translates :vimeo_url, backend: :container, column_name: 'custom_fields'
end
```

## Expected Behavior
I expect to use `Content.new(vimeo_url: 'xxxxx')` or `cnt_instance.vimeo_url = 'xxxxx'`

## Actual Behavior
I get this error Mobility::Plugins::Backend::InvalidOptionKey (These are not valid jsonb backend keys: column_suffix.)
It seems to mix the default backend options and the specific one

Best regards,
Kevyn

**Comment by shioyama** [comment]
Thanks for the issue report! So this is not quite a "bug", more of an edge case that most people would not encounter :joy: 

Basically, when you set defaults like this:

```ruby
backend :jsonb, column_suffix: '_translations'
```

... they are mixed in every time you call `translates`. When you pass `backend: :container` you override the `backend` key, but the `column_suffix` key is still there and gets passed to `translates`. 

_However_, the `container` backend doesn't have a key called `column_suffix` so it complains about it.

You can get around this by defining a custom translations class instead of using `translates`, like this:

```ruby
class ContainerTranslations < Mobility::Translations
  plugins do
    backend :container
    column_name 'custom_fields'
  end
end

class Foo
  include ContainerTranslations.new(:vimeo_url
end
```

That should work, but note that none of the other plugin settings in `Mobility.configure` will be included here, so you'll need to duplicate them.

**Comment by shioyama** [comment]
Oh actually, there is another way to do this which is not documented but might be simpler:

```ruby
  translates :vimeo_url, backend: [:container, { column_name: 'custom_fields' }]
```

In this case, you pass the backend name _and_ its options to `backend`, which will completely override any defaults for the backend. I believe this should work.

**Comment by shioyama** [comment]
Ok I'm calling this a bug because I think it's natural to expect this to work. I think it can be fixed for the release of 1.1.

**Comment by kevynlebouille** [comment]
Indeed, I expected to override default backend (and options) per `translates` to change backend for specific use

Thanks,
Kevyn

**Comment by shioyama** [comment]
This has been released as [1.0.4](https://rubygems.org/gems/mobility/versions/1.0.4), @kevynlebouille can you check the new version works as expected?

**Comment by kevynlebouille** [comment]
Hi Chris,

Thank you for this release! It works as expected.

Magnifique :)

Kevyn
